File: fintrack-backend/src/main/java/com/fintrack/fintrackbackend/dto/ExpenseRequest.java
package com.fintrack.fintrackbackend.dto;

import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;

public class ExpenseRequest {

    @NotNull(message = "Amount is required")
    private BigDecimal amount;

    @NotNull(message = "Category ID is required")
    private Integer categoryId;

    private String description;

    // Getters & Setters
    public BigDecimal getAmount() { return amount; }
    public void setAmount(BigDecimal amount) { this.amount = amount; }

    public Integer getCategoryId() { return categoryId; }
    public void setCategoryId(Integer categoryId) { this.categoryId = categoryId; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
}

File: frontend/app/src/main/java/pe/fintrack/mobile/ui/theme/data/model/ExpenseRequest.kt
package pe.fintrack.mobile.ui.theme.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import java.math.BigDecimal

@JsonClass(generateAdapter = true)
data class ExpenseRequest(
    @Json(name = "amount") val amount: BigDecimal,
    @Json(name = "categoryId") val categoryId: Int,
    @Json(name = "description") val description: String?
)